ytdl_nfo/__init__.py

```diff
@@ -15,14 +15,16 @@ def main():
         file = Ytdl_nfo(args.input, args.extractor)
         file.process()
     else:
-        for filename in os.listdir(args.input):
-            if filename.endswith('.json'):
-                print(f'Processing {filename} with extractor {args.extractor}')
-                file = Ytdl_nfo(os.path.join(args.input,  filename), args.extractor)
-                file.process()
+        for root, dirs, files in os.walk(args.input):
+            for file_name in files:
+                file_path = os.path.join(root, file_name)
+                if file_name.endswith('.info.json'):
+                    print(f'Processing {file_path} with extractor {args.extractor}')
+                    file = Ytdl_nfo(file_path, args.extractor)
+                    file.process()
```

Comment:
I needed recursive traversal so I can easily apply to my whole library at once, which is arranged e.g. `videos/[uploader]/[title]`

Reply:
Also, youtube-dl metadata files _should_ specifically end in `.info.json`. It might not be sensible to assume any and all json files in a directory are from youtube-dl. (Even so it might be a good idea to better handle the case where a given json file turned out not to contain valid metadata, even an `.info.json` one)